inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/openapi/OpenInlongTenantController.java

```diff
@@ -0,0 +1,83 @@
+import org.apache.inlong.manager.pojo.tenant.InlongTenantInfo;
+import org.apache.inlong.manager.pojo.tenant.InlongTenantPageRequest;
+import org.apache.inlong.manager.pojo.tenant.InlongTenantRequest;
+import org.apache.inlong.manager.pojo.user.LoginUserUtils;
+import org.apache.inlong.manager.service.operationlog.OperationLog;
+import org.apache.inlong.manager.service.tenant.InlongTenantService;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/openapi")
+@Api(tags = "Open-InlongTenant-API")
+public class OpenInlongTenantController {
+
+    @Autowired
+    private InlongTenantService tenantService;
+
+    @RequestMapping(value = "/tenant/get/{name}", method = RequestMethod.GET)
+    @ApiOperation(value = "Get inlong tenant by name")
+    @ApiImplicitParam(name = "name", dataTypeClass = String.class, required = true)
+    public Response<InlongTenantInfo> get(@PathVariable String name) {
+        return Response.success(tenantService.getByName(name));
+    }
+
+    @RequestMapping(value = "/tenant/save", method = RequestMethod.POST)
+    @OperationLog(operation = OperationType.CREATE)
+    @ApiOperation(value = "Save inlong tenant")
+    public Response<Integer> save(@Validated @RequestBody InlongTenantRequest request) {
+        return Response.success(tenantService.save(request));
+    }
+
+    @RequestMapping(value = "/tenant/list", method = RequestMethod.POST)
+    @ApiOperation(value = "List tenant by paginating")
+    public Response<PageResult<InlongTenantInfo>> listByCondition(@RequestBody InlongTenantPageRequest request) {
+        return Response.success(tenantService.listByCondition(request, LoginUserUtils.getLoginUser()));
+    }
+
+    @RequestMapping(value = "/tenant/update", method = RequestMethod.POST)
+    @OperationLog(operation = OperationType.UPDATE)
+    @ApiOperation(value = "Update inlong tenant")
+    public Response<Boolean> update(@Validated(UpdateByIdValidation.class) @RequestBody InlongTenantRequest request) {
+        return Response.success(tenantService.update(request));
+    }
+
+    @RequestMapping(value = "/tenant/delete/{name}", method = RequestMethod.GET)
+    @ApiOperation(value = "Delete inlong tenant by name")
+    @ApiImplicitParam(name = "name", dataTypeClass = String.class, required = true)
+    public Response<Boolean> delete(@PathVariable String name) {
+        return Response.success(tenantService.delete(name));
```

Comment:
Check whether the tenant has been refered by business before deleting.

Reply:
create a new issue to fix it